### PR TITLE
RD-4493 follow up PR

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
This is a follow-up PR to #1935 with additions and changes to tests.

## Screenshots / Videos
N/A

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [ ] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
25 May 2022:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2794/pipeline

Jenkins system tests:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2784/pipeline

 

## Documentation
N/A

### CODE REVIEW

**Comment by leon-good-life** [comment]
TODO: remove `.DS_STORE` add it to gitignore. (It is an annoying file that is being created every time on Mac that stores information about how I viewed a folder last time)

File: test/cypress/integration/widgets/deployment_button_configuration_spec.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+describe('should allow configuring deployment button', () => {
+    const widgetId = 'deploymentButton';
+    const resourcePrefix = 'deploy_test_';
+    const labelsBlueprint = `${resourcePrefix}labels`;
+    // const withoutLabelsBlueprint = `${resourcePrefix}without_labels`;
+
+    before(() => {
+        cy.activate('valid_trial_license').usePageMock('deploymentButton').mockLogin();
+        // cy.uploadBlueprint('blueprints/without_labels.zip', withoutLabelsBlueprint);
+        cy.uploadBlueprint('blueprints/labels.zip', labelsBlueprint);
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply test/cypress/integration/widgets/deployment_button_configuration_spec.ts]: I addressed it: [d81333d](https://github.com/cloudify-cosmo/cloudify-stage/pull/1978/commits/d81333d11386d61584ed48d7135451b17436e73b)

File: test/cypress/integration/widgets/deployment_button_configuration_spec.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        cy.setSearchableDropdownConfigurationField(widgetId, 'Icon', icon);
+        cy.get('button i').should('have.class', icon);
+    });
+
+    it('label filter rules', () => {
+        cy.get('div.deployBlueprintModal').within(() => {
+            openDropdown('blueprintName').within(() => {
+                cy.get('[role="listbox"] > *').should('not.have.length', 1);
+            });
+            cy.get('.actions > .ui:nth-child(1)').click();
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply test/cypress/integration/widgets/deployment_button_configuration_spec.ts]: I addressed it: [88b6e68](https://github.com/cloudify-cosmo/cloudify-stage/pull/1978/commits/88b6e683b4164b806c7e507682ae7ea63211197d)

File: test/cypress/integration/widgets/deployment_button_configuration_spec.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    it('icon', () => {
+        const icon = 'play';
+        cy.setSearchableDropdownConfigurationField(widgetId, 'Icon', icon);
+        cy.get('button i').should('have.class', icon);
+    });
+
+    it('label filter rules', () => {
+        cy.get('div.deployBlueprintModal').within(() => {
+            openDropdown('blueprintName').within(() => {
+                cy.get('[role="listbox"] > *').should('not.have.length', 1);
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply test/cypress/integration/widgets/deployment_button_configuration_spec.ts]: I addressed it: [3a89f70](https://github.com/cloudify-cosmo/cloudify-stage/pull/1978/commits/3a89f70e2591e685eed73d12759b6c7806139393)

File: test/cypress/integration/widgets/deployment_button_configuration_spec.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
+describe('should allow configuring deployment button', () => {
+    const widgetId = 'deploymentButton';
+    const resourcePrefix = 'deploy_test_';
+    const labelsBlueprint = `${resourcePrefix}labels`;
+    // const withoutLabelsBlueprint = `${resourcePrefix}without_labels`;
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply test/cypress/integration/widgets/deployment_button_configuration_spec.ts]: I addressed it: [8827aca](https://github.com/cloudify-cosmo/cloudify-stage/pull/1978/commits/8827acac4d478ff3488fc6b99b9cd7e44dc74acd)

**Comment by leon-good-life** [comment]
I started `deployment_button_configuration_spec` in Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2830/pipeline

**Comment by leon-good-life** [comment]
`deployment_button_configuration_spec` passed.

**Comment by leon-good-life** [comment]
**I started two tests together on Jenkins:**
1. `deployment_button_spec`
2. `deployment_button_configuration_spec`
Link: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2834/pipeline

**Comment by leon-good-life** [comment]
I fixed `allows to deploy a blueprint` test by removing `async` and `await` from `drilldownPage.ts`.

I started `deployment_button_spec` and `deployment_button_configuration_spec` in Jenkins:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2865/pipeline

**Comment by leon-good-life** [comment]
Hi @kubama , @qooban approved this PR, but asked that both `deployment_button_spec` and `deployment_button_configuration_spec ` will pass on Jenkins. 

The only change since then is that I removed `async` and `await` from `drilldownPage.ts`

**Comment by kubama** [comment]
This is a change that affects the application globally, so please verify that all system tests are passing.

**Comment by leon-good-life** [comment]
I started all system tests on Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2866/pipeline

**Comment by leon-good-life** [comment]
tests are passing except the tests that are failing on master branch.

**Comment by kubama** [comment]
> tests are passing except the tests that are failing on master branch.

This is not the case, there are 21 passing tests in `blueprints_spec.ts` on master and only 16 on this branch. Apparently something may have been broken by the change.

**Comment by leon-good-life** [comment]
I reverted my recent change and started all system tests again:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2871/pipeline

**Comment by leon-good-life** [comment]
@kubama Is it okay to merge this PR without fixing `allows to deploy a blueprint` in this PR.

This PR is already fixes one test and refactors code.

**Review by kubama — APPROVED** [review]

**Pull request merged**